pennylane/qnode.py

```diff
@@ -184,10 +184,12 @@ def __init__(
         self.qfunc_output = None
         # store the user-specified differentiation method
         self.diff_method = diff_method
+        self.diff_method_change = False
 
         self._tape, self.interface, self.device, tape_diff_options = self.get_tape(
             device, interface, diff_method
         )
+        self.set_best_diff_method(tape_diff_options)

```

Comment:
Looks nice! I'd suggest altering this slightly though, for clarity. Perhaps having something like:
```python
# if diff_method is best, then set it to the actual diff method being used
if self.diff_method == "best":
    self.diff_method = self._get_best_diff_method(tape_diff_options)
```

updating the method below accordingly.

This would:
1. make it easier to follow what's being done
2. avoid having the helper method changing attributes, leaving that to the `__init__` method instead
3. only calling the helper method when necessary
4. adding the `_`, indicating that this is an internal method, not be used outside of the class

Reply:
Makes sense. Implemented.

Reply:
Great, thanks @ankit27kh! This method can now be a `@staticmethod`, since it doesn't use `self`.

Reply:
Yes, I have changed that.